### DISCUSSION TITLE
Integer scale and keep aspect ratio are not working.

### DISCUSSION THREAD

**Discussion by febs** [eduardofilo/RG350_adam_image#298] [RetroArch topics]
### What happened?

I enter any GB game, then enter retroarch through select+X.

I go to settings-> video -> scaling and enable integer scale.

Game is still full screen.

Moreover, if I override the config, when I restart the game, the setting for integer scaling is still enabled (although ignored; but at least it appears "on" in retroarch); but the "Keep aspect ratio" is back to "off". It looks like only the first "override" is applied.

Thanks for any help.

### Device

RG280M v1.1 (without HDMI)

### Version

v2.1

**Comment by febs** [comment]
Addendum.
I noticed that if I use the "darken" filter, the proportions do work and eventually also the integer scaling is respected - which by the way makes the screen to small for gb I realized but this is another story.

Apparently most filters do ignore scaling.

**Comment by febs** [comment]
Or, if I just remove any filters of course the aspect ratio is respected.

**Comment by eduardofilo** [comment]
It's not that these options don't work or are not persisted in the overrides, it's that RetroArch has a rather cumbersome way of managing them. Read this to see if it helps:
https://github.com/eduardofilo/RG350_adam_image/wiki/En:-7.-FAQ#q3-why-are-the-override-settings-in-retroarch-not-preserved

**Comment by febs** [comment]
Thanks for this! I learnt something about persistence in this powerful, busy mess that's retroarch. :-D

But the main problem I wanted to note is that, persistence apart, those settings are just _plain ignored as long as there's any filter active that isn't "darken.filt"_

With no filters or with (the indeed quite useless) "darken.filt" the preserve aspect ratio option is respected. Otherwise it's totally ignored.

Is there any way around this?

I'd really like to play GB in its original aspect ratio.
Many thanks,

**Reply by eduardofilo** [reply]
I still think you are missing some control over how you have made the configurations at Core, Content or ROM level. I haven't had any problems setting up the system with for example the `Dot_Matrix_3x_Dark_Grid` filter and the Keep Aspect Ratio option activated.

Read carefully what I said in the FAQ I mentioned before about the "economy of settings" that RetroArch does when generating the entries inside the cfg files. Surely you have an override at the content level that is complicating things to you.
